### ISSUE TITLE
from_server does not find an existing OpenFFWorkflow

### ISSUE THREAD

**Issue by ChayaSt** [MolSSI/QCFractal#91]
**Describe the bug**
When I try to retrieve an existing workflow by calling `collections.OpenFFWorkflow.from_server(name='chemper2_rdkit', client=client)` of a collection that I had already registered, I get the following error:
```
KeyError                                  Traceback (most recent call last)
<ipython-input-3-b4da8d7fca6e> in <module>
----> 1 wf_2 = portal.collections.OpenFFWorkflow.from_server(name='chemper2_rdkit', client=client)

~/src/molssi/QCFractal/qcfractal/interface/collections/collection.py in from_server(cls, client, name)
    121             raise KeyError("Warning! `{}: {}` not found.".format(class_name, name))
    122 
--> 123         return cls.from_json(tmp_data["data"][0], client=client)
    124 
    125     @classmethod

~/src/molssi/QCFractal/qcfractal/interface/collections/collection.py in from_json(cls, data, client)
    168         print('data: {}'.format(data))
    169         # Allow PyDantic to handle type validation
--> 170         return cls(name, client=client, **data)
    171 
    172     def to_json(self, filename=None):

~/src/molssi/QCFractal/qcfractal/interface/collections/openffworkflow.py in __init__(self, name, options, client, **kwargs)
     39         # Expand options
     40         if options is None:
---> 41             raise KeyError("No record of OpenFFWorkflow {} found and no initial options passed in.".format(name))
     42         super().__init__(name, client=client, **options, **kwargs)
     43 

KeyError: 'No record of OpenFFWorkflow chemper2_rdkit found and no initial options passed in.'
```
But when I look at `client.get_collection("OpenFFWorkflow", 'chemper2_rdkit', full_return=True)`
I see that the workflow does exist:
```
{'meta': {'errors': [],
  'n_found': 1,
  'success': True,
  'error_description': False,
  'missing': []},
 'data': [{'name': 'chemper2_rdkit',
   'collection': 'openffworkflow',
   ...
```
But since `options` is `None` [here](https://github.com/MolSSI/QCFractal/blob/master/qcfractal/interface/collections/openffworkflow.py#L22-L41) It raises a `KeyError`. 
**To Reproduce**
```
# set up a client
client = portal.FractalClient("https://localhost:7777/", verify=False)
# register workflow
with open('example_workflow.json') as f:
    wf_json = json.load(f)['example']['fragmenter']
wf = portal.collections.OpenFFWorkflow(name='example', client=client, options=wf_json)
# new workflow by getting an existing one from the server
wf_2 = portal.collections.OpenFFWorkflow.from_server(name='example', client=client)
```
The example workflow is [here](https://github.com/openforcefield/fragmenter/blob/workflow/examples/example_workflow.json)

@dgasmith

**Comment by ChayaSt** [comment]
In addition, the `DataModel` class is missing `torsiondrive_input`.

**Comment by dgasmith** [comment]
Do you like the `from_server` model vs the `client.get_collection` model? I am not sure we want to support both in the long run.

**Comment by ChayaSt** [comment]
If I have to choose one, I prefer the  `client.get_collection` model. 
If I don't set `full_return` to True then I still get the same error because it can't initialize a new `OpenFFWorkflow` object with `options=None`.

**Comment by ChayaSt** [comment]
Another thing that's weird is that I added `torsiondrive_input` to `DataModel` (in a separate branch) but when I check `wf.get_options('torsiondrive_input')` I just get an empty dictionary while the other  fields (`enumerate_states`, `enumerate_fragments`) do have their options.

**Comment by dgasmith** [comment]
Is this pulling from the MolSSI server that we spun up? As keys were changed when we pulled your PR in the two are no longer correctly mapping to each other. As we do not yet have versioning of these schema implemented this is to be expected. Rolling back ~20 commits or so will have the server we spun up for you working. We can look at migrating the current collection for you.

+1 for `client.get_collection` which I believe I prefer over getting it from the object.

**Comment by ChayaSt** [comment]
No. I spun up a local server to test the newer constraints scans and am getting this error. The Collection was registered with the newer keys.

**Comment by ChayaSt** [comment]
When pulling from the MolSSI server, I get this error:
wf = molssi_client.get_collection(collection_name="chemper1_psi4", collection_type='OpenFFWorkflow')
```
KeyError                                  Traceback (most recent call last)
<ipython-input-4-4bd350d3e983> in <module>
----> 1 wf = molssi_client.get_collection(collection_name="chemper1_psi4", collection_type='OpenFFWorkflow')

~/src/molssi/QCPortal/qcportal/client.py in get_collection(self, collection_type, collection_name, full_return)
    304             # If nothing found
    305             if len(r.json()["data"]):
--> 306                 return collection_factory(r.json()["data"][0], client=self)
    307             else:
    308                 return None

~/src/molssi/QCPortal/qcportal/collections/collection_utils.py in collection_factory(data, client)
     63         raise KeyError("Attempted to create Collection of unknown type '{}'.".format(data["collection"]))
     64 
---> 65     return __registered_collections[data["collection"].lower()].from_json(data, client=client)

~/src/molssi/QCPortal/qcportal/collections/collection.py in from_json(cls, data, client)
    159         missing = req_fields - data.keys()
    160         if len(missing):
--> 161             raise KeyError("For class {} the following fields are missing {}.".format(class_name, missing))
    162         name = data.pop('name')
    163         # Allow PyDantic to handle type validation

KeyError: "For class openffworkflow the following fields are missing {'optimization_static_options', 'torsiondrive_static_options'}."
```

**Comment by dgasmith** [comment]
Right, the current code will not work from the MolSSI server as the OpenFFWorkflow collection object's definition changed when we added your changes from #88.

@Lnaden Can verify, but I think the signature `def __init__(self, name, options=None, client=None, **kwargs)` is incorrect compared to `def __init__(self, name, options=None, client=None, **kwargs)`. So we will need:

```
def __init__(self, name, client=None, **kwargs):
    super().__init__(name, client=client, **kwargs)
```

**Comment by ChayaSt** [comment]
This change works. 
Should I open a PR with this change?

**Comment by ChayaSt** [comment]
fix with #92
